api/envoy/api/v2/route/route.proto

```diff
@@ -152,14 +152,22 @@ message VirtualHost {
   google.protobuf.UInt32Value per_request_buffer_limit_bytes = 18;
 }
 
+// A filter-defined action type.
+message FilterAction {
+  oneof action {
+    google.protobuf.Any filter_action = 1;
```

Comment:
You don't need `Struct` any more, this has been marked for deprecation everywhere thanks to great nice work by @yanavlasov.

Reply:
How is this going to work for clients like gRPC, which can't handle arbitrary unknown proto types in `Any` fields?  Is there a design doc for this?

Reply:
In this scenario, we will embed `Struct` within `Any`, take a look at https://github.com/envoyproxy/envoy/issues/4475 for the design discussion.

Reply:
How does the client tell the server that it needs the server to embed a `Struct` inside the `Any`?  We had been talking about introducing a client capability to indicate that requirement.  Was that added somewhere?

Reply:
This needs adding still, @yanavlasov is working on a PR to add client capabilities.

Reply:
> How does the client tell the server that it needs the server to embed a `Struct` inside the `Any`? We had been talking about introducing a client capability to indicate that requirement. Was that added somewhere?

That's the client capability to indicate, but that's orthogonal to the `oneof` vs `Struct-in-Any` here. The following question is still valid:
How does the client tell the server that it needs the server to send a `Struct` instead of the `Any`?

Reply:
I think that if we had the client capability, that would solve this problem, as I described in https://github.com/envoyproxy/envoy/issues/4475#issuecomment-507439716.  But until we have the client capability, I don't see how we can get rid of the `Struct`, because we'd have no mechanism to ensure that the client gets a `Struct` inside of the `Any`.

Reply:
@markdroth That still doesn't make difference between `Struct` vs `Struct-in-Any`? For an (extreme) example, if the management server always send `Struct-in-Any` then client capability isn't needed.

Reply:
Ah, I see -- you're saying that even with separate `Struct` and `Any` fields, the server still has to know which one to populate for a given client, so not having a separate `Struct` field isn't any worse than that, even without the client capability.

Okay, I'll buy that argument.  I've removed the `Struct` field here.